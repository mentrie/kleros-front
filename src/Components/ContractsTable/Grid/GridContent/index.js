import React, { Component } from 'react'
import _ from 'lodash'
import { connect } from 'react-redux'
import { withRouter, Link } from 'react-router-dom'
import 'babel-polyfill'
import FontAwesome from 'react-fontawesome'
import { getContracts } from '../../../../redux/contracts/action-creators'
import { truncateText } from '../../../../helpers/truncateText'
import './GridContent.css'

export class GridContent extends Component {
  componentWillMount () {
    this.props.getDataContracts()
  }

  render () {
    const { hasErrored, isFetching, contracts = [] } = this.props
    if (hasErrored) {
      return <p>Sorry! There was an error loading the balance</p>
    }

    if (isFetching) {
      return (
        <div className='GridContent-container'>
          <div className='items loader'>
            <div className='linear-background-100' />
            <div className='linear-background-90' />
            <div className='linear-background-90' />
          </div>
        </div>
      )
    }

    return (
      <div className='GridContent-container'>
        <div className='items'>
          {
            _.isEmpty(contracts) &&
            <div className='items-row'>
              <div className='item item-no-contracts'>
                You have no contracts.
              </div>
            </div>
          }
          {
            contracts.filter(contract => {
              return contract.arbitrator === process.env.REACT_APP_ARBITRATOR_ADDRESS
            }).map(contract =>
              (
                <Link key={contract.address} to={`contract-summary/${contract.address}`}>
                  <div className='items-row'>
                    <div className='item-contract-hash'>{contract.address}</div>
                    <div className='item item-party-a'>{truncateText(contract.partyA, 10)}</div>
                    <div className='item item-party-b'>{truncateText(contract.partyB, 10)}</div>
                    <div className='item item-rule'>
                      <FontAwesome name='circle-thin' />
                    </div>
                  </div>
                </Link>
              )
            )
          }
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => {
  return {
    contracts: state.contracts.contracts,
    hasErrored: state.contracts.failureDisputes,
    isFetching: state.contracts.requestDisputes
  }
}

const mapDispatchToProps = dispatch => {
  return {
    getDataContracts: () => dispatch(getContracts())
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(GridContent))
