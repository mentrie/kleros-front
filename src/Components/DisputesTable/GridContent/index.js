import React, { Component } from 'react'
import _ from 'lodash'
import { connect } from 'react-redux'
import { withRouter, Link } from 'react-router-dom'
import 'babel-polyfill'
import { getDisputes } from '../../../redux/disputes/action-creators'
import { truncateText } from '../../../helpers/truncateText'
import { STATUS_TO_STATE } from '../../../constants'
import './GridContent.css'

export class GridContent extends Component {
  componentWillMount () {
    this.props.getDataDisputes()
  }

  render () {
    const { hasErrored, isFetching, disputes, baseLink, filterFunction } = this.props
    if (hasErrored) {
      return <p>Sorry! There was an error loading the disputes</p>
    }

    if (isFetching) {
      return (
        <div className='GridContent-container'>
          <div className='items loader'>
            <div className='linear-background-100' />
            <div className='linear-background-90' />
            <div className='linear-background-90' />
          </div>
        </div>
      )
    }

    const filteredDisputes = filterFunction(disputes)

    return (
      <div className='GridContent-container'>
        <div className='items'>
          {
            _.isEmpty(disputes) &&
            <div className='items-row'>
              <div className='item item-no-disputes'>
                You have no disputes.
              </div>
            </div>
          }
          {
            filteredDisputes.filter(dispute => {
              return dispute.arbitratorAddress === process.env.REACT_APP_ARBITRATOR_ADDRESS
            }).map(dispute =>
              (
                <Link key={dispute.arbitrableContractAddress} to={`${baseLink}/${dispute.arbitrableContractAddress}`}>
                  <div className='items-row'>
                    <div className='item item-project'>
                      <div className='item-title'>{ truncateText(dispute.description ? dispute.description : 'unavailable', 35) }</div>
                      <div className='item-category'>{ dispute.category }</div>
                    </div>
                    <div className='item item-deadline'>{ dispute.deadline }</div>
                    <div className='item item-case_id'>{ truncateText(dispute.hash, 10) }</div>
                    <div className='item item-status'>{ STATUS_TO_STATE[dispute.arbitrableContractStatus] }</div>
                  </div>
                </Link>
              )
            )
          }
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => {
  return {
    disputes: state.disputes.disputes,
    hasErrored: state.disputes.failureDisputes,
    isFetching: state.disputes.requestDisputes
  }
}

const mapDispatchToProps = dispatch => {
  return {
    getDataDisputes: () => dispatch(getDisputes())
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(GridContent))
