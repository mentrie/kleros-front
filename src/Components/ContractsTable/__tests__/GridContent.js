import React from 'react'
import { GridContent } from '../Grid/GridContent'

describe('GridContent Exists', () => {
  const props = {
    getDataContracts: jest.fn(),
    contracts: [{address: '0xwere', partA: '', partB: '0x2343'}]
  }
  const component = mount(<GridContent {...props} />)
  test('should render component properly', () => {
    expect(component).toMatchSnapshot()
  })

  test('should render contracts when present', () => {
    expect(component.props().contracts.length).toBe(1)
  })

  test('should render empty div when there\'s no contract', () => {
    const props = {
      getDataContracts: jest.fn()
    }
    const component = shallow(<GridContent {...props} />)
    expect(component.find('.item-no-contracts').length).toBe(1)
  })
})
