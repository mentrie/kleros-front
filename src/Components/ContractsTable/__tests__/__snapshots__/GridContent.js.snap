// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridContent Exists should render component properly 1`] = `
<GridContent
  contracts={
    Array [
      Object {
        "address": "0xwere",
        "partA": "",
        "partB": "0x2343",
      },
    ]
  }
  getDataContracts={[Function]}
>
  <div
    className="GridContent-container"
  >
    <div
      className="items"
    />
  </div>
</GridContent>
`;
